File: java/components/validation-ctt/HEAD/ctt_testcases_v14.12.00a/src/com/at4wireless/alljoyn/core/about/AboutAnnouncement.java
/*
 * Copyright AllSeen Alliance. All rights reserved.
 *
 *    Permission to use, copy, modify, and/or distribute this software for any
 *    purpose with or without fee is hereby granted, provided that the above
 *    copyright notice and this permission notice appear in all copies.
 *
 *    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
 *    WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
 *    MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
 *    ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
 *    WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
 *    ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
 *    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
 */
package com.at4wireless.alljoyn.core.about;

import java.util.Map;

import org.alljoyn.bus.AboutObjectDescription;
import org.alljoyn.bus.Variant;
import org.alljoyn.services.common.BusObjectDescription;

// TODO: Auto-generated Javadoc
/**
 * This class contains details of the announcement broadcast by About feature
 * about the device/app and list of supported interfaces.
 * 
 */
public class AboutAnnouncement
{
    
    /** The service name. */
    private String serviceName;
    
    /** The port. */
    private short port;
    
    /** The version. */
    private int version;
    
    /** The object descriptions. */
    private AboutObjectDescription[] objectDescriptions;
    
    /** The about data. */
    private Map<String, Variant> aboutData;
    
    /**
     * @param serviceName
     *            the bus unique name of the peer exposing the service
     * @param port
     *            the port on which the app will listen
     * @param objectDescriptions2
     *            object paths and the list of all interfaces available at the
     *            given object path
     * @param aboutData
     *            app/device details
     */
    public AboutAnnouncement(String serviceName, int version2, short port, AboutObjectDescription[] objectDescriptions2, Map<String, Variant> aboutData)
    {
    	this.version=version2;
        this.serviceName = serviceName;
        this.port = port;
        this.objectDescriptions = objectDescriptions2;
        this.aboutData = aboutData;
    }

    /**
     * @return the bus unique name of the peer exposing the service
     */
    public String getServiceName()
    {
        return serviceName;
    }
    
    /**
     * Gets the version.
     *
     * @return the version
     */
    public int getVersion()
    {
        return version;
    }

    /**
     * @return port on which the app will listen
     */
    public short getPort()
    {
        return port;
    }

    /**
     * @return array of {@code BusObjectDescription} which contains object paths
     *         and the list of all interfaces available at the given object path
     */
    public AboutObjectDescription[] getObjectDescriptions()
    {
        return objectDescriptions;
    }

    /**
     * @return app/device details
     */
    public Map<String, Variant> getAboutData()
    {
        return aboutData;
    }
}

File: cpp/components/validation-ctt/HEAD/ctt_testcases/AboutAnnouncement.cpp
/******************************************************************************
* Copyright AllSeen Alliance. All rights reserved.
*
*    Permission to use, copy, modify, and/or distribute this software for any
*    purpose with or without fee is hereby granted, provided that the above
*    copyright notice and this permission notice appear in all copies.
*
*    THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
*    WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
*    MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
*    ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
*    WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
*    ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
*    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
******************************************************************************/
#include "stdafx.h"
#include "AboutAnnouncement.h"

using namespace ajn;
using namespace std;

AboutAnnouncement::AboutAnnouncement(AJ_PCSTR t_ServiceName,
    uint16_t t_Version, uint16_t t_Port, const MsgArg& t_ObjectDescription, 
    const MsgArg& t_AboutData) : 
    m_Version(t_Version), m_ServiceName(t_ServiceName), m_Port(t_Port)
{

    m_ObjectDescription = new AboutObjectDescription();
    m_ObjectDescription->CreateFromMsgArg(t_ObjectDescription);

    m_AboutData = new AboutData(t_AboutData);
}

string AboutAnnouncement::getServiceName() const
{
    return m_ServiceName;
}

uint16_t AboutAnnouncement::getPort() const
{
    return m_Port;
}

AboutObjectDescription* AboutAnnouncement::getObjectDescriptions() const
{
    return m_ObjectDescription;
}

uint16_t AboutAnnouncement::getVersion() const
{
    return m_Version;
}

AboutData* AboutAnnouncement::getAboutData() const
{
    return m_AboutData;
}